src/clj_jgit/porcelain.clj

```diff
@@ -787,11 +787,11 @@
          .call
          blame-result)))
 
-(defn get-blob-id
-  [repo commit path]
+(defn ^org.eclipse.jgit.lib.ObjectId get-blob-id
+  [^Git repo ^org.eclipse.jgit.revwalk.RevCommit commit ^String path]
   (let [tree-walk (TreeWalk/forPath (.getRepository repo) path (.getTree commit))]
     (when tree-walk
-      (.getObjectId tree-walk 0))))
+      (.getObjectId ^org.eclipse.jgit.treewalk.TreeWalk tree-walk 0))))
```

Comment:
This hint doesn't seem to be required, at least I got no reflection warning leaving it out. If it is required it's probably better to put the hint into the let where the var is defined.